tap_postgres/client.py

```diff
@@ -134,5 +198,8 @@ def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
             if start_val:
                 query = query.filter(replication_key_col >= start_val)
 
+        if self._MAX_RECORDS_LIMIT is not None:
```

Comment:
Whats this for?

Reply:
Not sure what the full scope of use cases is but just can be overridden to limit the number of rows fetched. It's used in [Tap.run_connection_test()](https://github.com/meltano/sdk/blob/ee570e0068980613731c71ff6188da4e3b1941e2/singer_sdk/tap_base.py#L190). And [here's an issue](https://github.com/meltano/sdk/issues/1110) where the Meltano team seems to have wanted it in there recently.

Reply:
This is in the SDK now we should remove this from here I believe.

Reply:
It's in the SDK in [SQLStream.get_records](https://github.com/meltano/sdk/blob/ee570e0068980613731c71ff6188da4e3b1941e2/singer_sdk/streams/sql.py#L197)d, which we override here. I don't think we want to arbitrarily drop this functionality. Maybe chat more about this on Slack? I might be missing something